homeassistant/components/fitbit/sensor.py

```diff
@@ -472,6 +472,11 @@ def device_state_attributes(self):
     def update(self):
         """Get the latest data from the Fitbit API and update the states."""
         if self.resource_type == "devices/battery" and self.extra:
+            new_dev_status = self.client.get_devices()
+            for dev_extra in new_dev_status:
+                if dev_extra.get("id") == self.extra.get("id"):
+                    self.extra = dev_extra
```

Comment:
You could make this shorter by using the filter function:
```suggestion
            device_id = self.extra.get("id")
            self.extra = list(filter(lambda device: device.get("id") == device_id, registered_devs))[0]
```

Reply:
Ok I implemented all the suggested changes, ran tests to make sure it works and uploaded the new version.